unit/util/sharing_map.cpp

```diff
@@ -156,6 +156,10 @@ TEST_CASE("Sharing map interface", "[core][util]")
     REQUIRE(sm.has_key("i"));
     REQUIRE(sm.has_key("j"));
     REQUIRE(!sm.has_key("k"));
+
+    cbmc_invariants_should_throwt invariants_throw;
+
+    REQUIRE_THROWS(sm.insert("i", "4"));
```

Comment:
I would suggest to also require a particular exception type (using `REQUIRE_THROWS_AS`). See other unit tests for examples.